src/librustc_lexer/src/lib.rs

```diff
@@ -14,157 +14,20 @@
 // We want to be able to build this crate with a stable compiler, so no
 // `#![feature]` attributes should be added.
 
+mod literal;
+mod token;
 mod cursor;
+mod utils;
 pub mod unescape;
 
-use crate::cursor::{Cursor, EOF_CHAR};
-
-/// Parsed token.
-/// It doesn't contain information about data that has been parsed,
-/// only the type of the token and its size.
-pub struct Token {
-    pub kind: TokenKind,
-    pub len: usize,
-}
+pub use self::token::{Token, TokenKind};
+pub use self::literal::{LiteralKind, Base};
+pub use self::utils::{is_whitespace, is_id_start, is_id_continue, strip_shebang};
```

Comment:
If I were splitting this into pieces, I'd probably instead moved `advanced_token` and friends to `imp` module, as those are private implementation details, and don't have to be in the crate root.

Reply:
On the other hand, when you open the directory with files, you see the `token.rs` and can open it to see **only tokens definition**, without any other information related to parsing.

Reply:
I judge based on my own experience: since I'm only getting familiar with the compiler, I explore modules by opening it and attempting to start from the most important thing that I can find. For example, in `libsyntax/parse` it's simple: you open `token.rs` and see the types of tokens produced by the module, and that's it: you're ready to explore the rest of the crate.